turbine/src/retransmit_stage.rs

```diff
@@ -403,16 +537,15 @@ impl RetransmitStage {
             CLUSTER_NODES_CACHE_NUM_EPOCH_CAP,
             CLUSTER_NODES_CACHE_TTL,
         );
-
-        let mut stats = RetransmitStats::new(Instant::now());
-
         let mut rng = rand::thread_rng();
+        let mut stats = RetransmitStats::new(Instant::now());
+        let mut addr_cache = AddrCache::with_capacity(/*capacity:*/ 4);
         let mut shred_deduper = ShredDeduper::new(&mut rng, DEDUPER_NUM_BITS);
 
         let thread_pool = {
-            // Using clamp will panic if less than 8 sockets are provided
+            // Using clamp will panic if max < min.
             #[allow(clippy::manual_clamp)]
-            let num_threads = get_thread_count().min(8).max(retransmit_sockets.len());
+            let num_threads = get_thread_count().min(12).max(retransmit_sockets.len());
```

Comment:
Do you recall how much of the perf gains came from the rest of the PR vs. from bumping the threadpool size ? Definitely don't spend time collecting data if you don't; I'm just curious as I could see both aspects yielding gains independently

Reply:
This one increases throughput, does little for latency (which is where speculative computation is benefitial). So they both provide benefits, just different ones.